### PR TITLE
Magic byte signing and verification

### PR DESCRIPTION
**Pull request by jmininger**

### CODE REVIEW

File: src/subcommands/util.rs
Comment (jhartzell42) [inline src/subcommands/util.rs]: You mean with_name(binary_hex_arg_name)

File: src/subcommands/util.rs
Comment (jhartzell42) [inline src/subcommands/util.rs]: Same here.

File: src/subcommands/util.rs
Comment (jhartzell42) [inline src/subcommands/util.rs]: Isn't this precluded/handled already by what we have above? Can't submit a PR against upstream with a TODO.

**Review by jhartzell42 — CHANGES_REQUESTED** [review]

File: src/subcommands/util.rs
Comment (TheWaWaR) [inline src/subcommands/util.rs:333]: Have you considered the use case which don't want the magic bytes?
For example: a DApp want sign a message use `ckb-cli`.

Comment (mikereinhart) [reply src/subcommands/util.rs:333]: Are these two use cases different enough that they should be different subcommands? If there are messages with and without the magic byte it can potentially undermine the security benefits of this feature. Any information about this additional use case is helpful and appreciated!

Comment (TheWaWaR) [reply src/subcommands/util.rs:333]: The point is the message need to be signed is from a CKB transaction generated by the DApp. Typically, the message is  calculated by `blake2b(ckb_tx_hash + witnesses)` and the message need to put back into witness for type/lock script to verify. Just like a message from normal CKB transaction which transfer some capacity from one address to another use default lock script.

Comment (TheWaWaR) [reply src/subcommands/util.rs:333]: Maybe better add a flag `--no-magic-bytes` for security consideration, then the default behave is changed, so it's a breaking change.

Comment (mikereinhart) [reply src/subcommands/util.rs:333]: For now I think it makes sense for us to merge this PR if we feel it adequately satisfies magic byte message signing. That way we can move forward with dev/QA of this feature on the Ledger device.

We should do another internal PR to account for this DApp use case and the `--no-magic-bytes` flag (after specification for it is finalized). Both of these features should be upstreamed to CKB-CLI in a single PR when ready.

Comment (TheWaWaR) [reply src/subcommands/util.rs:333]: I'm OK with the way move forward. Just one thing need to mention, I think the magic-bytes value deserve a `const` key word (only defined once in the whole project).

Comment (jhartzell42) [reply src/subcommands/util.rs:333]: We also consolidated sign-data with sign-message. Does that make sense to you?

Comment (TheWaWaR) [reply src/subcommands/util.rs:333]: Yes, make sense to me.

**Comment by jhartzell42** [comment]
This PR is obsolete and a new ticket has been made with new requirements for an upstream PR. The requirements changed upon further conversation with the client.

**Comment by jmininger** [comment]
Superseded by https://github.com/obsidiansystems/ckb-cli/pull/21

**Pull request closed without merging**
